### PR TITLE
Update services "Application existence check" needs to be updated

### LINKED ISSUE #146: Update services "Application existence check" needs to be updated
Currently , for the following services , 
- /v1/update-all-ltps-and-fcs
- /v1/update-ltp
- /v1/update-fc
- /v1/update-fc-port
- /v1/delete-fc-port
- /v1/delete-ltp-and-dependents

before updating the data to the load file , server side implementation checks whether this is a "regarded application" or not. 
For this check , "originator" header from the request body is considered to check against the list of http-client-interface-configuration/application-name. 
In the acceptance testsuite , serveral testcases are failing as the originator for all the request is "postman". Also , there might be a possibility that an admin can perform this operations via swagger UI. Until and unless if it is a proper request from a server , in other scenarios there is a possibility that the real time value of the Originator depends on the situation. 

So , instead of considering the application-name from the Originator header , it shall be considered for the following services 
- /v1/update-all-ltps-and-fcs -From the request body , consider application-name from the http-server-capability
- /v1/update-ltp
  /v1/update-fc
  /v1/update-fc-port
  /v1/delete-fc-port
  /v1/delete-ltp-and-dependents 
For these services , currently , from the uuid , we are deriving the control-construt uuid to update the instance.
So , if a control-construct instance is available , that means its a proper request.
Also , since the security is enabled from the approved links from ALT , this request can be trusted too.

### PR DESCRIPTION
**Pull request by vanithavalluripalli9**
Fixes #146

### CODE REVIEW

**Comment by vanithavalluripalli9** [comment]
Reviewer is @DanaSunal

File: server/service/individualServices/ControlConstructService.js [hunk 1]
```diff
@@ -93,20 +96,30 @@ class ControlConstrucService {
...
+     filter_path: "hits.hits",
+     body: {
+       "query": {
+         "term": {
+           "uuid": controlConstructUuid
+         }
+       }
+     }
+   })
+   if (Object.keys(res.body.hits.hits).length === 0) {
```
Comment (DanaSunal) [inline server/service/individualServices/ControlConstructService.js:117]: ES will return empty body object if no control-construct was found, which means this check will fail on 'cannot read property hits on undefined'.

File: server/service/IndividualServicesService.js [hunk 2]
```diff
@@ -219,15 +219,15 @@ exports.bequeathYourDataAndDie = function (body, user, originator, xCorrelator,
...
-      await checkApplicationExists(originator);
-
       /****************************************************************************************
        * Setting up required local variables from the request body
        ****************************************************************************************/
       let forwardingConstructUuid = body["fc-uuid"];
       let fcPortLocalId = body["fc-port-local-id"];
       let controlConstructUuid = figureOutControlConstructUuid(forwardingConstructUuid);
 
+      await checkApplicationExists(controlConstructUuid);
```
Comment (DanaSunal) [inline server/service/IndividualServicesService.js:229]: This is not needed here. The method 'getForwardingDomainFromControlConstruct' already does this check, no need to double check it.

File: server/service/IndividualServicesService.js [hunk 3]
```diff
@@ -261,14 +261,14 @@ exports.deleteFcPort = function (body, user, originator, xCorrelator, traceIndic
...
-      await checkApplicationExists(originator);
-
       /****************************************************************************************
        * Setting up required local variables from the request body
        ****************************************************************************************/
       let logicalTerminationPointUuid = body["uuid"];
       let controlConstructUuid = figureOutControlConstructUuid(logicalTerminationPointUuid);
 
+      await checkApplicationExists(controlConstructUuid);
+
```
Comment (DanaSunal) [inline server/service/IndividualServicesService.js:271]: No need for this here either, my PR #268 already takes care of that.

File: server/service/IndividualServicesService.js [hunk 4]
```diff
@@ -1145,15 +1145,15 @@ exports.updateAllLtpsAndFcs = async function (body, originator) {
...
-      await checkApplicationExists(originator);
-
       /****************************************************************************************
        * Setting up required local variables from the request body
        ****************************************************************************************/
       let forwardingConstruct = body;
       let forwardingConstructUuid = forwardingConstruct["uuid"];
       let controlConstructUuid = figureOutControlConstructUuid(forwardingConstructUuid);
 
+      await checkApplicationExists(controlConstructUuid);
```
Comment (DanaSunal) [inline server/service/IndividualServicesService.js:1155]: Same as with delete-fc-port.

File: server/service/IndividualServicesService.js [hunk 5]
```diff
@@ -1196,6 +1195,8 @@ exports.updateFcPort = function (body, user, originator, xCorrelator, traceIndic
       let fcPort = body["fc-port"];
       let controlConstructUuid = figureOutControlConstructUuid(forwardingConstructUuid);
 
+      await checkApplicationExists(controlConstructUuid);
```
Comment (DanaSunal) [inline server/service/IndividualServicesService.js:1198]: Same as with delete-fc-port.

File: server/service/IndividualServicesService.js [hunk 6]
```diff
@@ -1238,6 +1238,8 @@ exports.updateLtp = function (body, user, originator, xCorrelator, traceIndicato
       let logicalTerminationPointUuid = logicalTerminationPoint[onfAttributes.GLOBAL_CLASS.UUID];
       let controlConstructUuid = figureOutControlConstructUuid(logicalTerminationPointUuid);
 
+      await checkApplicationExists(controlConstructUuid);
```
Comment (DanaSunal) [inline server/service/IndividualServicesService.js:1241]: PR #267 takes care of this.

**Review by DanaSunal — CHANGES_REQUESTED** [review]

File: server/service/IndividualServicesService.js (hunk 2 above)
Comment (vanithavalluripalli9) [reply server/service/IndividualServicesService.js]: Done

File: server/service/IndividualServicesService.js (hunk 3 above)
Comment (vanithavalluripalli9) [reply server/service/IndividualServicesService.js]: Done

File: server/service/IndividualServicesService.js (hunk 4 above)
Comment (vanithavalluripalli9) [reply server/service/IndividualServicesService.js]: Done

File: server/service/IndividualServicesService.js (hunk 5 above)
Comment (vanithavalluripalli9) [reply server/service/IndividualServicesService.js]: Done

File: server/service/individualServices/ControlConstructService.js (hunk 1 above)
Comment (vanithavalluripalli9) [reply server/service/individualServices/ControlConstructService.js:117]: ES was returning body.hits.hits as empty object . where as body was containing hits object . So , even when controlconstruct was not present in ES , it was not throwing the error .

Comment (DanaSunal) [reply server/service/individualServices/ControlConstructService.js:117]: I see... when the filter path is hits.hits, it returns `{
  "hits" : {
    "hits" : [ ]
  }
}` when the filter path is hits.hits._source it returns an empty object. `{}`

File: server/service/individualServices/ControlConstructService.js [hunk 7]
```diff
@@ -12,6 +12,9 @@ const onfAttributes = require('onf-core-model-ap/applicationPattern/onfModel/con
 const onfFormatter = require('onf-core-model-ap/applicationPattern/onfModel/utility/OnfAttributeFormatter');
 const fileOperation = require('onf-core-model-ap/applicationPattern/databaseDriver/JSONDriver');
 const LayerProtocol = require('onf-core-model-ap/applicationPattern/onfModel/models/LayerProtocol');
+const { elasticsearchService, getIndexAliasAsync } = require('onf-core-model-ap/applicationPattern/services/ElasticsearchService');
+
+const ELASTICSEARCH_CLIENT_CC_UUID = "alt-2-0-1-es-c-es-1-0-0-000";
```
Comment (DanaSunal) [inline server/service/individualServices/ControlConstructService.js:17]: Use ElasticsearchPreparation.

**Review by DanaSunal — DISMISSED** [review]

**Pull request merged**